vizro-core/src/vizro/models/_page.py

```diff
@@ -103,10 +94,34 @@ def __init__(self, **data):
                 f"as the page title. If you have multiple pages with the same title then you must assign a unique id."
             ) from exc
 
+    def _get_page_actions_chains(self) -> List[ActionsChain]:
+        """Gets all ActionsChains present on the page."""
+        page_actions_chains = []
+
+        for model_id in model_manager._get_model_children(model_id=ModelID(str(self.id))):
+            model = model_manager[model_id]
+            if hasattr(model, "actions"):
+                page_actions_chains.extend(model.actions)
+
+        for control in self.controls:
+            if hasattr(control, "selector") and control.selector:
+                page_actions_chains.extend(control.selector.actions)
+
+        return page_actions_chains
```

Comment:
Maybe this can also make use of the recursive approach outlined before, as this looks again quite implementation specific

Reply:
I also agree, I will add TODOs where these improvements are mentioned so we can develop these when we decide which way ModelManager will go.